// Copyright (C) 2018 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package severity;
option java_package = "com.google.gapid.proto.service.severity";
option java_outer_classname = "SeverityLevel";
option go_package = "github.com/google/gapid/gapis/service/severity";

// Severity defines the severity of a logging message.
// The values must be identical to values in core/log/severity.go
enum Severity {
  // Verbose indicates extremely verbose level messages.
  VerboseLevel = 0;
  // Debug indicates debug-level messages.
  DebugLevel = 1;
  // Info indicates minor informational messages that should generally be
  // ignored.
  InfoLevel = 2;
  // Warning indicates issues that might affect performance or compatibility,
  // but could be ignored.
  WarningLevel = 3;
  // Error indicates non terminal failure conditions that may have an effect on
  // results.
  ErrorLevel = 4;
  // Fatal indicates a fatal error and the process should be terminated.
  FatalLevel = 5;
}
